### PR TITLE
Make batch segment allocation logs less noisy

### PR DESCRIPTION
**Pull request by kfaraz**
### Changes

- Change some batch allocation logs from info to debug as these are noisy and make debugging actual issues difficult

### CODE REVIEW

**Review by AmatyaAvadhanula — APPROVED** [review]

File: indexing-service/src/main/java/org/apache/druid/indexing/common/actions/SegmentAllocationQueue.java [hunk 1]
```diff
@@ -601,7 +599,7 @@ synchronized void handleResult(SegmentAllocateResult result, SegmentAllocateRequ
         emitTaskMetric("task/action/success/count", 1L, request);
         requestToFuture.remove(request).complete(result.getSegmentId());
       } else if (request.canRetry()) {
-        log.info(
+        log.debug(
```
Comment (abhishekagarwal87) [inline indexing-service/src/main/java/org/apache/druid/indexing/common/actions/SegmentAllocationQueue.java:602]: this should still be info.

File: indexing-service/src/main/java/org/apache/druid/indexing/common/actions/SegmentAllocationQueue.java [hunk 2]
```diff
@@ -355,7 +353,7 @@ private boolean processBatch(AllocateRequestBatch requestBatch)
 
     emitBatchMetric("task/action/batch/attempts", 1L, requestKey);
     emitBatchMetric("task/action/batch/runTime", (System.currentTimeMillis() - startTimeMillis), requestKey);
-    log.info("Successfully processed [%d / %d] requests in batch [%s].", successCount, batchSize, requestKey);
+    log.debug("Successfully processed [%d / %d] requests in batch [%s].", successCount, batchSize, requestKey);
```
Comment (abhishekagarwal87) [inline indexing-service/src/main/java/org/apache/druid/indexing/common/actions/SegmentAllocationQueue.java:356]: is it too noisy?

File: indexing-service/src/main/java/org/apache/druid/indexing/common/actions/SegmentAllocationQueue.java [hunk 3]
```diff
@@ -301,7 +299,7 @@ private void processBatchesDue()
       nextScheduleDelay = Math.max(0, maxWaitTimeMillis - timeElapsed);
     }
     scheduleQueuePoll(nextScheduleDelay);
-    log.info("Processed [%d] batches, next execution in [%d ms]", numProcessedBatches, nextScheduleDelay);
+    log.debug("Processed [%d] batches, next execution in [%d ms]", numProcessedBatches, nextScheduleDelay);
```
[the remark this answers is not in the document]
Comment (kfaraz) [reply indexing-service/src/main/java/org/apache/druid/indexing/common/actions/SegmentAllocationQueue.java:302]: Yes, this adds a log line typically every 0.5 s even when not doing anything.

File: indexing-service/src/main/java/org/apache/druid/indexing/common/actions/SegmentAllocationQueue.java (hunk 2 above)
Comment (kfaraz) [reply indexing-service/src/main/java/org/apache/druid/indexing/common/actions/SegmentAllocationQueue.java:356]: It is once per batch, so I guess we can go either way with this.

**Pull request merged**